Mypy failing on Python 3.10.7

Mypy is raising the following error on Actions for python 3.10:

Run mypy src --cache-dir mypy-cache-3.10
[2](https://github.com/pmgbergen/porepy/actions/runs/3124467882/jobs/5067855558#step:13:2) mypy src --cache-dir mypy-cache-3.10
[3](https://github.com/pmgbergen/porepy/actions/runs/3124467882/jobs/5067855558#step:13:3) shell: /usr/bin/bash -e {0}
[4](https://github.com/pmgbergen/porepy/actions/runs/3124467882/jobs/5067855558#step:13:4) env:
[5](https://github.com/pmgbergen/porepy/actions/runs/3124467882/jobs/5067855558#step:13:5) pythonLocation: /opt/hostedtoolcache/Python/3.10.7/x64
[6](https://github.com/pmgbergen/porepy/actions/runs/3124467882/jobs/5067855558#step:13:6) LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.10.7/x64/lib
[7](https://github.com/pmgbergen/porepy/actions/runs/3124467882/jobs/5067855558#step:13:8) /opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/numpy/__init__.pyi:642: error: Positional-only parameters are only supported in Python 3.8 and greater
[8](https://github.com/pmgbergen/porepy/actions/runs/3124467882/jobs/5067855558#step:13:9) Found 1 error in 1 file (errors prevented further checking)
[9](https://github.com/pmgbergen/porepy/actions/runs/3124467882/jobs/5067855558#step:13:10) Error: Process completed with exit code 2.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -27,7 +27,7 @@ jobs:
     # https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#specifying-a-python-version
     strategy:
       matrix:
-        python-version:  ["3.8", "3.9", "3.10.6"]
+        python-version:  ["3.8", "3.9", "3.10.6"]  # TODO: Change 3.10.6 to 3.10 when mypy error is fixed
       # Complete all versions in matrix even if one fails.
       fail-fast: false
 

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -27,7 +27,7 @@ jobs:
     # https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#specifying-a-python-version
     strategy:
       matrix:
-        python-version:  ["3.8", "3.9", "3.10"]
+        python-version:  ["3.8", "3.9", "3.10.6"]
       # Complete all versions in matrix even if one fails.
       fail-fast: false
 

